syntax = "proto2";

import "data.proto";

option cc_generic_services = true;


message GossipData {
  repeated Data ranges = 1; 
  optional string addr = 2; // we dont' need this, only for debug msg
}

service GossipService {
  rpc ManualGossip(EmptyMessage) returns (EmptyMessage);
  rpc PushData(GossipData) returns (EmptyMessage);
  rpc PullData(GossipData) returns (GossipData);
}
